src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3150,6 +3152,14 @@ private VisitArgumentResult VisitArgumentEvaluate(BoundExpression argument, RefK
             Debug.Assert(!this.IsConditionalState);
         }
 
+        private void CheckDisallowedNullAssignment(TypeWithState state, TypeWithAnnotations type, FlowAnalysisAnnotations annotations, Location location)
+        {
+            if (type.IsNullableType() && state.MayBeNull && ((annotations & FlowAnalysisAnnotations.DisallowNull) != 0))
```

Comment:
>type.IsNullableType() [](start = 16, length = 21)

Do we want to cover type parameters constrained to a nullable value type? This check will not cover them. Please add a test regardless.

Reply:
I didn't understand. 
I don't think we can constrain a type parameter to a nullable value type: `where T : U? where U : struct` is an error. (error CS0701: 'U?' is not a valid constraint. A type used as a constraint must be an interface, a non-sealed class or a type parameter.)

---
In reply to: [292606714](https://github.com/dotnet/roslyn/pull/36263#discussion_r292606714) [](ancestors = 292606714)

Reply:
Chuck gave me some hint how to do that. 

---
In reply to: [292632259](https://github.com/dotnet/roslyn/pull/36263#discussion_r292632259) [](ancestors = 292632259,292606714)

Reply:
I don't see a unit-test for that scenario.

---
In reply to: [292634369](https://github.com/dotnet/roslyn/pull/36263#discussion_r292634369) [](ancestors = 292634369,292632259,292606714)